internal/services/eventgrid/eventgrid_namespace_resource.go

```diff
@@ -0,0 +1,610 @@
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r EventGridNamespaceResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.NamespacesClient
+
+			id, err := namespaces.ParseNamespaceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(existing.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := EventGridNamespaceResourceModel{
+				Name:          id.NamespaceName,
+				ResourceGroup: id.ResourceGroupName,
+			}
+
+			if model := existing.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+
+				if model.Sku != nil {
+					state.Sku = string(pointer.From(model.Sku.Name))
+					state.Capacity = pointer.From(model.Sku.Capacity)
+				}
+				flattenedIdentity, err := identity.FlattenSystemAndUserAssignedMapToModel(model.Identity)
+				if err != nil {
+					return fmt.Errorf("flattening `identity`: %+v", err)
+				}
+				state.Identity = pointer.From(flattenedIdentity)
+				state.Tags = pointer.From(model.Tags)
+
+				if props := model.Properties; props != nil {
+					topicSpacesConfig, err := flattenTopicSpacesConfiguration(props.TopicSpacesConfiguration)
+					if err != nil {
+						return fmt.Errorf("flattening `topic_spaces_configuration`: %v", err)
+					}
+					state.TopicSpacesConfiguration = topicSpacesConfig
+					state.InboundIpRules = flattenInboundIPRules(props.InboundIPRules)
+					state.PublicNetworkAccess = string(pointer.From(props.PublicNetworkAccess))
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r EventGridNamespaceResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.NamespacesClient
+
+			id, err := namespaces.ParseNamespaceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r EventGridNamespaceResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return namespaces.ValidateNamespaceID
+}
+
+func expandInboundIPRules(input []InboundIpRuleModel) *[]namespaces.InboundIPRule {
+	if len(input) == 0 {
+		return nil
+	}
+
+	var ipRules []namespaces.InboundIPRule
+	for _, v := range input {
+		ipRules = append(ipRules, namespaces.InboundIPRule{
+			Action: pointer.To(namespaces.IPActionType(v.Action)),
+			IPMask: pointer.To(v.IpMask),
+		})
+	}
+	return &ipRules
+}
+
+func flattenInboundIPRules(ipRules *[]namespaces.InboundIPRule) []InboundIpRuleModel {
+	var output []InboundIpRuleModel
```

Comment:
Forgot to flag this in my previous review, there is a difference between declaring a slice like this and like
`output := make([]InboundIpRuleModel, 0)`

Which is that the declaration `var output []InboundIpRuleModel` evaluates to `nil` if nothing has been appended to it, whereas initialising it with `make` will result in an empty slice `[]` which is what we want to setting values into state.

Can you please update this, as well as all the other flatten functions to 
```suggestion
	output := make([]InboundIpRuleModel, 0)
```